### ISSUE TITLE
[dash] Add sticky y axis per chart

### ISSUE THREAD

**Issue by jlubken** [CodeForPhilly/chime#342]
The single max_y_axis isn't very useful across plots that have very different y ranges.

Add a "sticky" value for each plot that stores the y_scale of the plot.

This may mean that the charts must be generated (not plotted) much earlier than they are now, in order to extract the y_scale.

Look for a hidden streamlit input. Users don't want to set the value, they want to hold a plot constant between plot refreshes.

**Comment by adibiasio** [comment]
I'll try this issue!

**Comment by adibiasio** [comment]
There might be a way to preserve the coordinates of a users graph. If they zoom in so that the maximal y-axis is their desired value, like this...

![Screenshot (6)](https://user-images.githubusercontent.com/61572112/79376089-a4967f80-7f27-11ea-96cf-a1ba917821b9.png)

The zoomed coordinates could be stored and reapplied when they update any parameters. The only time the graph would be reset is if the user manually presses the "Reset Axes" Button

**Comment by adibiasio** [comment]
# Dash
For future reference, I had originally looked into customizing the dash mode bar. I would have added a custom button within the mode bar. This was the ideal solution, however, this functionality is only available for dash with plotly JavaScript, not python. Here is how to implement a custom mode bar button in JavaScript.

```JavaScript
var icon1 = {
  'width': 500,
  'height': 600,
  'path': 'M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z'
}

var colors = ['green', 'red', 'blue']
var data = [{
  mode: 'lines',
  y: [2, 1, 2],
  line: {color: colors[0], width: 3, shape: 'spline'}
}]

var layout = {
  title: 'add mode bar button with custom icon',
  modebardisplay: false}

var config = {
  modeBarButtonsToAdd: [
    {
      name: 'color toggler',
      icon: icon1,
      click: function(gd) {
        var newColor = colors[Math.floor(3 * Math.random())]
        Plotly.restyle(gd, 'line.color', newColor)
      }},
    {
      name: 'button1',
      icon: Plotly.Icons.pencil,
      direction: 'up',
      click: function(gd) {alert('button1')
    }}],
  modeBarButtonsToRemove: ['pan2d','select2d','lasso2d','resetScale2d','zoomOut2d']}

Plotly.newPlot('myDiv', data, layout, config)
```

![Screenshot (8)](https://user-images.githubusercontent.com/61572112/79602052-ce7eac00-80b7-11ea-9a95-9922fef9ca7b.png)


Code From: https://plotly.com/javascript/configuration-options/#add-buttons-to-modebar
Here is another tutorial with more detail: https://www.somesolvedproblems.com/2018/10/how-to-customize-plotlys-modebar.html

This would be a much cleaner way of extending graph functionality, as the mode bar is hidden when not in use. For now, adding a button under the graph is fine, but like the sidebar, if too many features & buttons are added, it could start to fill up the page and make the app harder to use. If dash implements this feature in python or if we could embed JavaScript for the graphs, we could move the Lock Zoom button into the mode bar. Here is how the Lock Zoom feature is currently implemented.

![image](https://user-images.githubusercontent.com/61572112/79602803-2c5fc380-80b9-11ea-9c76-228ea829ff59.png)

When the Lock Zoom button is on, the graph retains its zoomed coordinates while the data is changed (parameter updates). When it is off, the graph will automatically set the axes to best fit all of the data. The Lock Zoom Button is in the ON position by default.


<p align="center">
  <img src="https://user-images.githubusercontent.com/61572112/79603264-fd961d00-80b9-11ea-9c74-f85798ed1ad8.png">
</p>

# Streamlit
I have not started working on this feature for streamlit yet

**Comment by BrianThomasRoss** [comment]
Closed with #557
